### PR TITLE
Unskip tests for Windows in `test_output`

### PR DESCRIPTION
**Pull request by czgdp1807**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tests/test_output.py [hunk 1]
```diff
@@ -337,15 +335,15 @@ def __repr__(self):
         out_str = proc.stdout.read().decode("ascii")
     else:
         out_str = proc.stderr.read().decode("ascii")
-    print(out_str)
```
Comment (pcmoritz) [inline python/ray/tests/test_output.py:340]: extra print

File: python/ray/tests/test_output.py [hunk 2]
```diff
@@ -337,15 +335,15 @@ def __repr__(self):
...
 
+    out_str = "".join(out_str.splitlines())
     assert "hi" in out_str, out_str
     assert "(Actor1 pid=" in out_str, out_str
     assert "bye" in out_str, out_str
     assert re.search("Actor2 pid=.*init", out_str), out_str
     assert not re.search("ActorX pid=.*init", out_str), out_str
     assert re.search("ActorX pid=.*bye", out_str), out_str
-    assert not re.search("Actor2 pid=.*bye", out_str), out_str
+    assert re.search("Actor2 pid=.*bye", out_str), out_str
```
Comment (pcmoritz) [inline python/ray/tests/test_output.py:346]: Hmm interesting, why the change here? cc @rkooo567

Comment (czgdp1807) [reply python/ray/tests/test_output.py:346]: After removing escape sequences from `out_str`, `re.search("Actor2 pid=.*bye", out_str)` gives `True`. The reason for this, IMO, is that due to presence of `Actor2\r` in `out_str`, `re.search` was giving `False`. However, now, since these escape sequences are removed the above test works fine. Is it something that shouldn't happen?

File: python/ray/tests/test_output.py (hunk 1 above)
Comment (czgdp1807) [reply python/ray/tests/test_output.py:340]: Should I add it back or keep it removed. I believe it's a debug print that was left behind.

File: python/ray/tests/test_output.py (hunk 2 above)
Comment (czgdp1807) [reply python/ray/tests/test_output.py:346]: Tests seem to be passing with this change.

Comment (pcmoritz) [reply python/ray/tests/test_output.py:346]: I think you are right, the test was probably wrong here. Let's see if somebody else has an opinion here and if not let's merge it :)

**Review by pcmoritz — APPROVED** [review]
